Meta/CMake/code_generators.cmake

```diff
@@ -0,0 +1,55 @@
+#
+# Functions for generating sources using host tools
+#
+
+function(compile_gml source output string_name)
+    set(source ${CMAKE_CURRENT_SOURCE_DIR}/${source})
+    add_custom_command(
+        OUTPUT ${output}
+        COMMAND ${SerenityOS_SOURCE_DIR}/Meta/text-to-cpp-string.sh ${string_name} ${source} > ${output}.tmp
```

Comment:
Technically, this could be replaced with a CMake script to make this less platform dependent. This script is fairly trivial as well:
```cmake
# Inputs: ${input} - path to file to read, ${output} - path to output to, ${string_name} - variable name
cmake_minimum_required(VERSION 3.16)

file(READ "${input}" content)

# Escape semicolons, so they are preserved as-is
string(REPLACE ";" "\\;" content "${content}")
# Turn each line into a list element
string(REPLACE "\n" ";" content "${content}")

set(result "\
extern const char ${string_name}[];
const char ${string_name}[] = R\"~~~(
")

foreach(line IN LISTS content)
  if(NOT line STREQUAL "" AND NOT line MATCHES "^ *#")
    string(APPEND result "${line}\n")
  endif()
endforeach()

string(APPEND result ")~~~\";\n")

file(WRITE "${output}" "${result}")
```

Reply:
Hrm. I'm not sure that eliminating the host platform dependency of

- having bash, and `/usr/bin/env`
- having grep that supports the regex '^ *#'

is of immediate concern to the project? 

I think until we figure out a reasonable way to create our cross-compilers for a Windows or non-posix host, (and have actual users/maintainers for such platforms), we'll pass on this particular cmake-ism.. the shell script is like 10 lines, and this script is about the same length as the current solution. Changing BuildIt.sh and BuildClang.sh to be platform independent would be a more interesting platform independence project imo 🤷‍♂️ .

Reply:
Yes, it was just a comment that I had to get out of my system. Technically, there is nothing that prevents Serenity from being compiled on platforms other than *nix.

Reply:
I know that feel. Also true, nothing preventing it assuming one can create the cross-compiler. Especially once I figure out the proper toolchain setup in my superbuild branch. Speaking of which, are we good on this PR? Feels like it's bout that time to get this in and iterate further on some future PRs.